src/compiler/compile/nodes/shared/Expression.ts

```diff
@@ -361,7 +360,7 @@ export default class Expression {
 					names.forEach(name => {
 						const dependencies = template_scope.dependencies_for_name.get(name);
 						if (dependencies) {
-							dependencies.forEach(name => traced.add(name));
+							dependencies.forEach(traced.add, traced);
```

Comment:
i don't understand why this change, you sure the inline function is the bottleneck, and the new code is significantly more performant than the existing one?